api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceServiceImpl.java

```diff
@@ -255,6 +259,68 @@ private void deleteUserRecentlyModifiedResourceEntry(
     }
   }
 
+  @Override
+  public List<DbUserRecentlyModifiedResource> findAllRecentlyModifiedResourcesByUser(long userId) {
+    try {
+      return DbRetryUtils.executeAndRetry(
+          () -> populateRecentlyModifiedResources(userId), Duration.ofSeconds(1), 5);
+    } catch (InterruptedException e) {
+      throw new ServerErrorException(
+          "Unable to find Recently Modified Resources for user" + userId);
+    }
+  }
+
+  public List<DbUserRecentlyModifiedResource> populateRecentlyModifiedResources(long userId) {
+    List<DbUserRecentlyModifiedResource> userList =
+        userRecentlyModifiedResourceDao.findDbUserRecentResourcesByUserId(userId);
+    userList.stream()
```

Comment:
I tried using @JOINCOLUMN, @WHERE in hibernate to populate cohort/conceptSet and dataset directly. That did not work, open for suggestions  for how to populate the objects rather than using switch cases

Reply:
I don't know Hibernate well enough to help here